pkg/volume/csi/csi_attacher.go

```diff
@@ -445,10 +386,11 @@ func (c *csiAttacher) Detach(volumeName string, nodeName types.NodeName) error {
 	}
 
 	klog.V(4).Info(log("detacher deleted ok VolumeAttachment.ID=%s", attachID))
-	return c.waitForVolumeDetachment(volID, attachID)
+	_, err := c.waitForVolumeDetachment(volID, attachID)
+	return err
 }
 
-func (c *csiAttacher) waitForVolumeDetachment(volumeHandle, attachID string) error {
+func (c *csiAttacher) waitForVolumeDetachment(volumeHandle, attachID string) (string, error) {
```

Comment:
Returning `string` looks useless, caller ignores it. It applies also to `waitForVolumeDetachmentInternal` below - nobody needs attachID return value during detach.

Reply:
Fixed